slang.h

```diff
@@ -1327,14 +1327,22 @@ extern "C"
 
     /** Add a distinct translation unit to the compilation request
 
-    `name` is optional.
+    `name` is optional. 
     Returns the zero-based index of the translation unit created.
     */
     SLANG_API int spAddTranslationUnit(
         SlangCompileRequest*    request,
         SlangSourceLanguage     language,
         char const*             name);
 
+    
+    /** Set a default module name. Translation units will default to this module name if one is not
+    passed. If not set each translation unit will get a unique name. 
+    */
+    SLANG_API void spSetDefaultModuleName(
+        SlangCompileRequest*    request,
+        const char* defaultModuleName);
```

Comment:
I'm not sure this makes sense to have as an API.
If users want to name their modules, then the `name` parameter to `spAddTranslationUnit` is the most direct way to do it.

More importantly, it doesn't make sense to have multiple translation units in a compile request creating modules with the *same* name. If the user wants that, then the right answer would be to have a *single* translation unit with multiple source files.

Having this entry point seems redundant in the single-translation-unit case, and potentially harmful in the multiple-translation-unit case.